src/main/java/de/dataelementhub/rest/controller/v1/ImportController.java

```diff
@@ -0,0 +1,225 @@
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.util.UriComponents;
+import org.springframework.web.util.UriComponentsBuilder;
+
+@RestController
+@RequestMapping("/v1")
+public class ImportController {
+
+  public String importDirectory = System.getProperty("java.io.tmpdir") + "/uploads";
+
+  private final ImportService importService;
+
+  @Autowired
+  public ImportController(ImportService importService) {
+    this.importService = importService;
+  }
+
+  /**
+   * Get an overview about all user imports and their status.
+   */
+  @Order(SecurityProperties.BASIC_AUTH_ORDER)
+  @GetMapping(value = "/import")
+  public ResponseEntity<List<ImportInfo>> allImports() {
+    Integer userId = DataElementHubRestApplication.getCurrentUser().getId();
+    List<ImportInfo> importInfoList = importService.allImports(userId);
+    return new ResponseEntity<List<ImportInfo>>(importInfoList, HttpStatus.OK);
+  }
+
+  /**
+   * Receive import files, start import process and return import id.
+   */
+  @Order(SecurityProperties.BASIC_AUTH_ORDER)
+  @PostMapping(consumes = {"multipart/form-data", "application/json"}, value = "/import")
+  public ResponseEntity<String> importFiles(@RequestBody List<MultipartFile> file,
+      @RequestParam("namespaceUrn") String namespaceUrn,
+      UriComponentsBuilder uriComponentsBuilder) {
+    Integer namespaceIdentifier = Integer.valueOf(namespaceUrn.split(":")[1]);
+    Integer userId = DataElementHubRestApplication.getCurrentUser().getId();
+    String timestamp = new Timestamp(System.currentTimeMillis())
+        .toString().replaceAll("[ \\.\\-\\:]", "_");
+    if (!DaoUtil.checkGrants(namespaceIdentifier, userId,
+        Arrays.asList(AccessLevelType.ADMIN, AccessLevelType.WRITE))) {
+      return new ResponseEntity<>(
+          "Only users with WRITE or ADMIN Grant can import to this namespace.",
+          HttpStatus.UNAUTHORIZED);
+    }
+    if (file == null) {
+      return new ResponseEntity<>(
+          "Uploaded File is not Valid.",
+          HttpStatus.BAD_REQUEST);
+    }
+    int importId = 0;
+    try {
+      importId = importService
+          .generateImportId(namespaceUrn, userId, file, importDirectory, timestamp);
+    } catch (IOException e) {
+      return new ResponseEntity<>(HttpStatus.UNPROCESSABLE_ENTITY);
```

Comment:
Does the excepion hold some more information about the cause of the problem? if so...I think it would be good to include it in the response or at least log it

Reply:
Done